node_modules/col-whiteboards/lib/api.js

```diff
@@ -755,7 +756,24 @@ var whiteboardAddActivity = function(socket, elements, callback) {
       AnalyticsAPI.track(socket.ctx.user, 'Add whiteboard element', AnalyticsAPI.getWhiteboardElementProperties(element, socket.whiteboard, elements.length));
     });
 
-    return callback();
+    async.each(elements, function(element, done) {
```

Comment:
I'm a little bit concerned about the `ActivitiesAPI.createActivity` function being executed in parallel, as students could be adding many assets at once. I believe that function adds the activity and recalculates and caches the total score for that user, and I wonder if that could suffer from a race condition.